histomicstk/FeatureExtraction.py

```diff
@@ -129,6 +187,325 @@ def FeatureExtraction(Label, I, K=128, Fs=6, Delta=8):
+        MeanIntensity, MeanMedianDifferenceIntensity,
+        MaxIntensity, MinIntensity, StdIntensity
+    )
+
+    return Output
+
+
+def GetPixCoords(Binary, bounds):
+    """
+    Get global coords of object extracted from tile
+    """
+    coords = np.where(Binary == 1)
+    coords = np.asarray(coords)
+    coords[0] = np.add(coords[0], bounds[0])
+    coords[1] = np.add(coords[1], bounds[2])
+    coords = coords.T
+
+    return coords
+
+
+def GetBounds(bbox, delta, N):
+    """
+    Returns bounds of object in global label image.
+    """
+    bounds = np.zeros(4, dtype=np.uint8)
+    bounds[0] = max(0, math.floor(bbox[0] - delta))
+    bounds[1] = min(N-1, math.ceil(bbox[0] + bbox[2] + delta))
+    bounds[2] = max(0, math.floor(bbox[1] - delta))
+    bounds[3] = min(N-1, math.ceil(bbox[1] + bbox[3] + delta))
+
+    return bounds
+
+
+def InterpolateArcLength(X, Y, L):
+    """
+    Resamples boundary points [X, Y] at L total equal arc-length locations.
+
+    Returns
+    -------
+    iX - L-length vector of horizontal interpolated coordinates with equal
+         arc-length spacing.
+    iY - L-length vector of vertical interpolated coordinates with equal
+         arc-length spacing.
+    """
+
+    # length of X
+    K = len(X)
+
+    # generate spaced points
+    Interval = np.linspace(0, 1, L)
+
+    # get segment lengths
+    Lengths = np.sqrt(
+        np.power(np.diff(X), 2) + np.power(np.diff(Y), 2)
+    )
+
+    # normalize to unit length
+    Lengths = Lengths / Lengths.sum()
+
+    # calculate cumulative length along boundary
+    Cumulative = np.hstack((0., np.cumsum(Lengths)))
+
+    # place points in 'Interval' along boundary
+    Locations = np.digitize(Interval, Cumulative)
+
+    # clip to ends
+    Locations[Locations < 1] = 1
+    Locations[Locations >= K] = K - 1
+    Locations = Locations - 1
+
+    # linear interpolation
+    Lie = np.divide(
+        (Interval - [Cumulative[i] for i in Locations]),
+        [Lengths[i] for i in Locations]
+    )
+
+    tX = np.array([X[i] for i in Locations])
+    tY = np.array([Y[i] for i in Locations])
+    iX = tX + np.multiply(
+        np.array([X[i+1] for i in Locations]) - tX, Lie
+    )
+    iY = tY + np.multiply(
+        np.array([Y[i+1] for i in Locations]) - tY, Lie
+    )
+
+    iXY = collections.namedtuple('iXY', ['iX', 'iY'])
+    Output = iXY(iX, iY)
+
+    return Output
+
+
+def FSDs(X, Y, K, Intervals):
+    """
+    Calculated FSDs from boundary points X,Y. Boundaries are resampled to have
+    K equally spaced points (arclength) around the shape. The curvature is
+    calculated using the cumulative angular function, measuring the
+    displacement of the tangent angle from the starting point of the boundary.
+    The K-length fft of the cumulative angular function is calculated, and
+    then the elements of 'F' are summed as the spectral energy over
+    'Intervals'.
+
+    Returns F - length(Intervals) vector containing spectral energy of
```

Comment:
Define return values like [this](https://github.com/DigitalSlideArchive/HistomicsTK/blob/master/histomicstk/ColorDeconvolution.py#L28-L41) in the doc string.

Also add docstrings for inputs/parameters. 

Do this for all functions.